### PR TITLE
[15.0][MIG] account_global_discount

### PR DESCRIPTION
**Pull request by kirca**

### CODE REVIEW

File: account_global_discount/views/report_account_invoice.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//div[@id='total']/div/table//tr[1]" position="before">
```
Comment (omar7r) [inline account_global_discount/views/report_account_invoice.xml]: With this position in report, this information only shows when invoice already paid, in addtion, it shows after total amount. You should change it by: `//div[@id='total']/div/table/t[@t-set='tax_totals'] position="before"`

**Review by omar7r — CHANGES_REQUESTED** [review]

File: account_global_discount/i18n/pt.po [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+#~ msgstr "Montante descontado"
+
+#~ msgid "Tax"
+#~ msgstr "Imposto"
+
+#~ msgid "<strong>Global Discounts</strong><br/>"
+#~ msgstr "<strong>Descontos Globais</strong><br/>"
+
+#~ msgid "Invoice Tax"
+#~ msgstr "Imposto de Fatura"
```
Comment (victoralmau) [inline account_global_discount/i18n/pt.po]:
```suggestion
```

File: account_global_discount/i18n/nl_NL.po [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+#~ msgstr "Basis korting"
+
+#~ msgid "Discounted amount"
+#~ msgstr "Kortingsbedrag"
+
+#~ msgid "Tax"
+#~ msgstr "Belasting"
+
+#~ msgid "Company related to this journal"
+#~ msgstr "Bedrijf gerelateerd aan journaal"
```
Comment (victoralmau) [inline account_global_discount/i18n/nl_NL.po]:
```suggestion
```

File: account_global_discount/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+#~ msgid "Tax"
+#~ msgstr "Impuesto"
+
+#~ msgid "<strong>Global Discounts</strong><br/>"
+#~ msgstr "<strong>Descuento Global</strong>"
+
+#, fuzzy
+#~ msgid "Invoice Tax"
+#~ msgstr "Factura"
```
Comment (victoralmau) [inline account_global_discount/i18n/es.po]:
```suggestion
```

File: account_global_discount/models/account_move.py [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+from odoo.tools import config
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # HACK: Looks like UI doesn't behave well with Many2many fields and
+    # negative groups when the same field is shown. In this case, we want to
+    # show the readonly version to any not in the global discount group.
+    # TODO: Check if it's fixed in future versions
```
Comment (victoralmau) [inline account_global_discount/models/account_move.py:15]: Have you checked if this is still happening in this version?

File: account_global_discount/models/account_move.py [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+        "line_ids.payment_id.state",
+        "invoice_global_discount_ids",
+        "global_discount_ids",
+    )
+    def _compute_amount(self):
+        """Modify totals computation for including global discounts."""
+        __ = super()._compute_amount()
+        for record in self:
+            record._compute_amount_one()
+        return __
```
Comment (victoralmau) [inline account_global_discount/models/account_move.py]:
```suggestion
        res = super()._compute_amount()
        for record in self:
            record._compute_amount_one()
        return res
```

File: account_global_discount/models/account_move.py [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+        """
+        moves = super().create(vals_list)
+        move_with_global_discounts = moves.filtered("global_discount_ids")
+        for move in move_with_global_discounts:
+            move.with_context(check_move_validity=False)._onchange_global_discount_ids()
+        return moves
+
+    def _check_balanced(self):
+        """Add the check of proper taxes for global discounts."""
+        __ = super()._check_balanced()
```
Comment (victoralmau) [inline account_global_discount/models/account_move.py]:
```suggestion
        res = super()._check_balanced()
```

File: account_global_discount/models/account_move.py [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+                for key in taxes_keys:
+                    if key == tuple(inv_line.tax_ids.ids):
+                        break
+                    elif set(key) & set(inv_line.tax_ids.ids) and test_condition:
+                        raise exceptions.UserError(
+                            _("Incompatible taxes found for global discounts.")
+                        )
+                else:
+                    taxes_keys[tuple(inv_line.tax_ids.ids)] = True
+        return __
```
Comment (victoralmau) [inline account_global_discount/models/account_move.py]:
```suggestion
        return res
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

File: account_global_discount/views/report_account_invoice.xml (hunk 1 above)
Comment (kirca) [reply account_global_discount/views/report_account_invoice.xml]: thanks

File: account_global_discount/models/account_move.py (hunk 5 above)
Comment (kirca) [reply account_global_discount/models/account_move.py:15]: It still seems to be an issue

**Review by victoralmau — APPROVED** [review]
Code review OK

**Comment by kirca** [comment]
Hi @amokaddem , why the additional commits? Last one breaks the pipeline

**Comment by kirca** [comment]
Hello @omar7r , could you check if it is good now? Thanks

**Review by omar7r — APPROVED** [review]
LGTM, tested

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

**Pull request merged**